retriever/lib/datapackage.py

```diff
@@ -356,7 +356,8 @@ def edit_dict(obj, tabwidth=0):
                         if del_val not in obj[key]:
                             print("Invalid value: Not found.")
                             continue
-                        print("Removed " + str(obj[key].pop(del_key)))
+                        obj[key].remove(del_val)
```

Comment:
Can you check the class of obj using print(class(obj));
This seems like a dictionary not a list.

Reply:
Yes, @henrykironde `obj` is a dictionary.

Reply:
I do not think remove will work since it is a dictionary

Reply:
@henrykironde but, `obj[key]` is a list, not a dictionary.

Reply:
@henrykironde have you tried removing an item from the `keywords` list of a dataset? e.g, the value `plants` from the database iris.

Reply:
@harshitbansal05, I have not tried running that part of code or tested it, I just read through the code. And it seems you are deleting something from a dictionary. I will have to run it and see what we are doing but it is clear that the initial value `del_key` was out of scope / not declared anywhere.

Reply:
I think @harshitbansal05 has it right in this correction. obj is a dictionary of key->list mappings, so obj[key] will return a list. the pop function is used only for indices, so we should use remove to delete the value.